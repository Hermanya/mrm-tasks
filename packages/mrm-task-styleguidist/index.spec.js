'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
}));

const fs = require.requireActual('fs');
const path = require('path');
const { omitBy } = require('lodash');
const getConfigGetter = require('../../mrm').getConfigGetter;
const vol = require('memfs').vol;
const task = require('./index');

const json = o => JSON.stringify(o, null, '  ');

afterEach(() => vol.reset());

it('should add a readme', () => {
	vol.fromJSON({
		[`${__dirname}/templates/styleguide.config.js`]: fs.readFileSync(
			path.join(__dirname, 'templates/styleguide.config.js')
		),
		'/package.json': json({
			name: 'unicorn',
		}),
	});

	task(getConfigGetter({}));

	expect(omitBy(vol.toJSON(), (v, k) => k.startsWith(__dirname))).toMatchSnapshot();
});

it.skip('should not install webpack when used with CRA', () => {});
