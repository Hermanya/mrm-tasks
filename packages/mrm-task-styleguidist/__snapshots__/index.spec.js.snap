// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a readme 1`] = `
Object {
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"styleguide\\": \\"styleguidist server\\",
    \\"styleguide:build\\": \\"styleguidist build\\",
    \\"start\\": \\"styleguidist server\\"
  }
}",
  "/styleguide.config.js": "module.exports = {
	components: 'src/components/**/[A-Z]*.js',
};
",
}
`;
